runtime/lua/vim/lsp/util.lua

```diff
@@ -1224,13 +1224,59 @@ end
 ---
 ---@param events (table) list of events
 ---@param winnr (number) window id of preview window
+---@param bufnrs (table) buffers where the preview window should remain visible
 ---@see |autocmd-events|
-function M.close_preview_autocmd(events, winnr)
-  if #events > 0 then
-    api.nvim_command("autocmd "..table.concat(events, ',').." <buffer> ++once lua pcall(vim.api.nvim_win_close, "..winnr..", true)")
+function M.close_preview_autocmd(events, winnr, bufnrs)
```

Comment:
Does this need to be a public function? grep shows it only used in util.lua, and somehow I strongly doubt plugins are using it.